web/experimental/inject_embed.dart

```diff
@@ -0,0 +1,72 @@
+import 'dart:html';
+
+import 'package:html_unescape/html_unescape.dart';
+
+// Change src to 'embed-new.html?fw=true for local development
+const iFrameLocation =
+    'https://dartpad.dartlang.org/experimental/embed-new.html?fw=true';
+
+/// Replaces all code snippets marked with the 'dartpad-embed' class with an
+/// instance of DartPad.
+void main() {
+  var hosts = querySelectorAll('.dartpad-embed');
+  for (var host in hosts) {
+    _injectEmbed(host);
+  }
+}
+
+/// Replaces [host] with an instance of DartPad as an embedded iframe.
+///
+/// Code snippets are assumed to be a div containing `pre` and `code` tags:
+///
+/// <div class="dartpad-embed">
+///   <pre>
+///     <code>
+///       void main() => print("Hello, World!");
+///     </code>
+///   </pre>
+/// </div>
+void _injectEmbed(DivElement host) {
+  if (host.children.length != 1) {
+    return;
```

Comment:
nit: Log something to the console prior to these returns, so people using the script will know why it's not working with their HTML.